### ISSUE TITLE
Emit proper error event to status when signature does not match

### ISSUE THREAD

**Issue by Gerrit91** [metal-stack/firewall-controller#93]
In the current implementation, when a user modifies the firewall resource without using a proper signature, the controller logs stack traces and also in the respective resource there is no hint that something is failing in the background.

Would be better if the controller adds a failure event for this resource such that it's possible to see when something like this is happening.

**Comment by GrigoriyMikhalkin** [comment]
Context: I tried to reproduce this problem locally, but was unsuccessful. Then @Gerrit91 and I looked into problem on an actual cluster and were able to reproduce it. But it was resolved after firewall-controller was reloaded.

First explanaition of this would be that with time channel becomes overloaded with events. And because default behavior is [DropIfChannelFull](https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/tools/record/event.go#L158) -- event broadcaster just drops these events. Problem with that is that it seems that only events with type `Warning` are dropped.

Another option would be some bug in [method that records events](
https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/tools/events/event_broadcaster.go#L168).

I suggest we postpone this issue until next occurrence of this problem. And if it occurs, i will add additional log messages in event recording logic to try to see, what exactly is happening.
